src/Core/Repositories/SqlServer/Repository.cs

```diff
@@ -43,16 +43,21 @@ public virtual async Task<T> GetByIdAsync(TId id)
             }
         }
 
-        public virtual async Task CreateAsync(T obj)
+        public virtual async Task<T> CreateAsync(T obj)
         {
             obj.SetNewId();
             using (var connection = new SqlConnection(ConnectionString))
             {
+                var parameters = new DynamicParameters();
+                parameters.AddDynamicParams(obj);
+                parameters.Add("Id", obj.Id, direction: ParameterDirection.InputOutput);
```

Comment:
What happens if the object, `T` has `Id` defined already here?

Reply:
Nothing. I couldn't find any documentation on this in Dapper, but my own debugging and the tests for entities that actually get IDs created by `SetNewId()` like User and Organization show that the flow is:
1. obj.Id gets set by `SetNewId()`
2. A template for parameters is created by `AddDynamicParams(obj)` that includes `obj.Id`
3. `parameters.Add(id)` creates a set-in-stone parameter, not a template like above, with the same Id value
4. Id is returned from the proc execution like normal.